hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFormatWriter.java

```diff
@@ -231,10 +235,15 @@ private void handleAppendExceptionOrRecoverLease(Path path, RemoteException e) t
       // last block. Find more information here : https://issues.apache.org/jira/browse/HDFS-6325
       log.warn("Failed to open an append stream to the log file. Opening a new log file..", e);
       // Rollover the current log file (since cannot get a stream handle) and create new one
-      this.logFile = logFile.rollOver(fs);
+      this.logFile = logFile.rollOver(fs, rolloverLogWriteToken);
+      createNewFile();
+    } else if (e.getClassName().contentEquals(AlreadyBeingCreatedException.class.getName())) {
+      log.warn("Another task executor writing to the same log file(" + logFile + ". Rolling over");
+      // Rollover the current log file (since cannot get a stream handle) and create new one
+      this.logFile = logFile.rollOver(fs, rolloverLogWriteToken);
```

Comment:
this seems like a good fix.. @n3nash fyi

Reply:
Yes, nice, thanks for catching this.